src/utils/SubscribableRequestQueue.ts

```diff
@@ -27,8 +27,14 @@ export default class SubscribableRequestQueue {
   /** Map from "inner" request (managed by `queue`) to "outer" promises generated per-subscriber. */
   private requests: Map<string, RequestSubscription[]>;
 
-  constructor(maxActiveRequests?: number) {
-    this.queue = new RequestQueue(maxActiveRequests);
+  constructor(maxActiveRequests?: number, maxLowPriorityRequests?: number);
+  constructor(inner: RequestQueue);
+  constructor(maxActiveRequests?: number | RequestQueue, maxLowPriorityRequests?: number) {
+    if (typeof maxActiveRequests === "number" || maxActiveRequests === undefined) {
+      this.queue = new RequestQueue(maxActiveRequests, maxLowPriorityRequests);
+    } else {
+      this.queue = maxActiveRequests;
+    }
```

Comment:
First time I've seen an overload like this, interesting. Maybe useful to add a docstring comment to explain that it can take in an existing RequestQueue as a parameter